lib/atomic-management.js

```diff
@@ -387,11 +405,110 @@ export default {
     atom.workspace.open(this.currentConfig);
   },
 
+  /**
+   * merge all config.cson detected in all open project.
+   * Upper projects take precedence over lower ones if there are conflicting entries
+   */
+  mergeConfigs() {
+    var mergedConfig = {};
+
+    this.projectConfigs.forEach(configName => {
+      let contents;
+      try {
+        if (configName.endsWith('.cson')) {
+          contents = CSON.readFileSync(configName);
+        } else if (configName.endsWith('.json')) {
+          contents = JSON.parse(fs.readFileSync(configName));
+        }
+      } catch (e) {
+        atom.notifications.addWarning(
+          [
+            'Unable to read supplied project specification file:\n\n',
+            `${configName}\n\n`,
+            'Please check the config file. Restoring global config.cson.'
+          ].join(''),
+          {
+            dismissable: true
+          }
+        );
+        console.log(e);
+        return;
+      }
+      if (contents === null) {
+        contents = {};
+      }
+      mergedConfig = mergeJSON.merge(contents,mergedConfig);
+      console.log(mergedConfig);
+      console.log("hahaha");
+      console.log(contents);
+    });
+
+    const currDisabled = this.getDisabledPackages(this.configContents);
+    const newDisabled = this.getDisabledPackages(mergedConfig);
+    if (this.configContents && !this.compareArrays(currDisabled, newDisabled)) {
+      this.askReload();
+    }
+
+    const installedPackageNames = new Set(
+      atom.packages.getAvailablePackageNames()
+    );
+    const notInstalledPackageNames = [];//newDisabled.filter(packageName => !installedPackageNames.has(packageName));
+    if("*" in mergedConfig){
+      for(var toplevel in mergedConfig){
+        for(var pname in mergedConfig[toplevel]){
+          if(pname != "core" &&
+             pname != "editor" &&
+             !installedPackageNames.has(pname)){
+            notInstalledPackageNames.push(pname);
+          }
+        }
+      }
+    } else {
+      for (var pname in mergedConfig) {
+        if (
+          pname !== 'core' &&
+          pname !== 'editor' &&
+          !installedPackageNames.has(pname)
+        ) {
+          notInstalledPackageNames.push(pname);
+        }
+      }
+    }
+
+    if (notInstalledPackageNames.length !== 0) {
+      this.askInstallPackages(notInstalledPackageNames);
+    }
+
+    this.config = this.projectConfigs[0];
+    this.currentConfig = this.projectConfigs[0];
+    this.configContents = mergedConfig;
+    if (mergedConfig) {
+      Object.keys(mergedConfig).forEach(namespace => {
```

Comment:
This part needs to also take into account when there are no namespaces (it starts with "core", "editor", etc)